### PR TITLE
Thumbnail product filtering

### PR DESCRIPTION
**Pull request by tokejepsen**
## Changelog Description
Port of https://github.com/ynput/OpenPype/pull/6197

### CODE REVIEW

File: client/ayon_core/plugins/publish/extract_thumbnail.py [hunk 1]
```diff
@@ -49,6 +50,8 @@ class ExtractThumbnail(pyblish.api.InstancePlugin):
     # attribute presets from settings
     oiiotool_defaults = None
     ffmpeg_args = None
+    subsets = []
```
Comment (iLLiCiTiT) [inline client/ayon_core/plugins/publish/extract_thumbnail.py:53]:
```suggestion
```

File: client/ayon_core/plugins/publish/extract_thumbnail.py [hunk 2]
```diff
@@ -103,6 +106,26 @@ def _main_process(self, instance):
...
             return
 
+        # We only want to process the subsets needed from settings.
+        def validate_string_against_patterns(input_str, patterns):
+            for pattern in patterns:
+                if re.match(pattern, input_str):
+                    return True
+            return False
+
+        product_names = self.subsets + self.product_names
```
Comment (iLLiCiTiT) [inline client/ayon_core/plugins/publish/extract_thumbnail.py:116]:
```suggestion
        product_names = self.product_names
```

File: client/ayon_core/version.py [hunk 3]
```diff
@@ -1,3 +1,3 @@
 # -*- coding: utf-8 -*-
 """Package declaring AYON core addon version."""
-__version__ = "0.2.1-dev.1"
+__version__ = "0.2.2"
```
Comment (iLLiCiTiT) [inline client/ayon_core/version.py:3]: Why?
```suggestion
__version__ = "0.2.1-dev.1"
```

Comment (tokejepsen) [reply client/ayon_core/version.py:3]: lol, dont know.

File: client/ayon_core/plugins/publish/extract_thumbnail.py [hunk 4]
```diff
@@ -103,6 +105,26 @@ def _main_process(self, instance):
...
+            return False
+
+        product_names = self.product_names
+        if product_names:
+            result = validate_string_against_patterns(
+                instance.data["subset"], product_names
+            )
+            if not result:
+                self.log.debug(
+                    "Subset \"{}\" did not match any valid subsets: {}".format(
```
Comment (iLLiCiTiT) [inline client/ayon_core/plugins/publish/extract_thumbnail.py:122]:
```suggestion
                    "Product name \"{}\" did not match settings filters: {}".format(
```

**Review by mkolar — APPROVED** [review]

**Pull request merged**